File: src/event/DeviceDataSet.php
<?php

namespace Ething\Event;

class DeviceDataSet extends AbstractResourceEvent {
	
	static public function emit(\Ething\Device\Device $resource, \stdClass $data){
		return new Signal('DeviceDataSet', array(
			'resource' => $resource->id(),
			'data' => $data
		));
	}
	
	static public function validate(array &$attributes, array $context){
		$context['onlyTypes'] = array('Device\\Device');
		return parent::validate($attributes, $context);
	}
	
	
	
}

File: ething/rule/event/DeviceDataSet.py


from .ResourceEvent import ResourceEvent

class DeviceDataSet(ResourceEvent):
	
	@classmethod
	def emit (cls, device, data):
		return super(DeviceDataSet, cls).emit(device, {
			'data' : data
		})
	
	@staticmethod
	def validate (attributes, context):
		return ResourceEvent.validate(attributes, context, onlyTypes=['Device'])
	
	
	
	

